packages/python-packages/apiview-stub-generator/apistub/_stub_generator.py

```diff
@@ -109,11 +108,12 @@ def __init__(self, **kwargs):
             logging.getLogger().setLevel(logging.DEBUG)
 
         # Extract package to temp directory if it is wheel or sdist
-        if self.pkg_path.endswith(".whl") or self.pkg_path.endswith(".zip"):
-            self.wheel_path, self.extras_require = self._extract_wheel()
+        if self.pkg_path.endswith((".whl", ".zip", ".tar.gz")):
+            self.wheel_path = self._extract_wheel()
+            self.namespace = self._get_whl_root_namespace(self.wheel_path)
         else:
-            # get extras_require from setup.py
-            self.wheel_path, self.extras_require = None, None
+            self.wheel_path = None
+            self.namespace = ""
```

Comment:
Did we want to do the path traversal here?

Reply:
Discussed OOB, but adding here as well: We need to extract the wheel before traversing.